pkg/specgen/generate/container_create.go

```diff
@@ -555,12 +555,27 @@ func createContainerOptions(rt *libpod.Runtime, s *specgen.SpecGenerator, pod *l
 	}
 	// Default used if not overridden on command line
 
-	if s.RestartPolicy != "" {
+	var (
+		restartPolicy string
+		retries       uint
+	)
+	// If the container is running in a pod, use the pod's restart policy for all the containers
+	if pod != nil {
+		podConfig, err := pod.Config()
```

Comment:
Just noticed: `pod.Config()` is a very expensive call as it's re-marshaling the entire config of the pod.

Non-blocking for this PR but please address it in a future one due avoid the perf regression: we should add `(*Pod).ConfigNoCopy()` as we did for containers.